### ISSUE TITLE
Let-in expressions remove surrounding empty lines

### ISSUE THREAD

**Issue by xeals** [nix-community/nixpkgs-fmt#218]
My understanding of the design goals is that this is a violation of goal (3). Otherwise, this is a consistency issue in that *only* `let ... in ...` expressions remove newlines.

## Input
```nix
a:

with a;

let

  b = { };

in

assert a ? foo;

a // b
```

## Output
```nix
a:

with a;
let

  b = { };

in
assert a ? foo;

a // b
```

## Desired output

```nix
a:

with a;

let

  b = { };

in

assert a ? foo;

a // b
```

**Comment by Rizary** [comment]
good point. I'll try to investigate the issue. I think we should adhere to the goal behind `nixpkgs-fmt`.

**Comment by flokli** [comment]
This is probably the same issue as `nixpkgs-fmt` currently removing the newline between the toplevel "function argument" and the let, as seen in https://github.com/NixOS/nixpkgs/pull/108991#discussion_r555720970.
